amivapi/auth/link_methods.py

```diff
@@ -0,0 +1,185 @@
+# -*- coding: utf-8 -*-
+#
+# license: AGPLv3, see LICENSE for details. In addition we strongly encourage
+#          you to buy us beer if we meet and you like the software.
+
+"""Functions adding permitted methods to '_links' part of response.
+
+Links are only added for resources using AmivTokenAuth.
+"""
+
+import json
+
+from flask import current_app, g
+from eve.auth import resource_auth
+
+from .auth import AmivTokenAuth, authenticate, check_if_admin
+
+
+def _get_item_methods(resource, item):
+    res = current_app.config['DOMAIN'][resource]
+    user = g.get('current_user')  # TODO: post_internal_problem
```

Comment:
Good catch! This todo can go away, since its ok like this. At the time I wasn't sure, but there is not always a user so using a `.get()` here is justified. (e.g. using an apikey)

But I am a little unhappy with how the link methods look. I have an idea how to code it nicer and have created an issue for this. But its not changing the result so i think its not critical.